### PR TITLE
Introduce DocFX for .NET API reference

### PR DESCRIPTION
**Pull request by mikhailshilkov**
Part of #2137, this PR introduces DocFX to generate .NET API Reference docs for our NuGet packages. For now, only the core "Pulumi" docs are generated.

### CODE REVIEW

**Comment by praneetloke** [comment]
@cnunciato wondering if we should merge this without anything linking to these auto-generated docs page yet?

@mikhailshilkov clicking on the "Improve this Doc" links navigates to some weird URL. Should we do something to enable that functionality? Or perhaps disable it?

Screenshot_2020-03-04 Class ComponentResource

Screenshot_2020-03-04 Class Output

**Comment by cnunciato** [comment]
@praneetloke I'm not sure I'd merge it without something linking to it, but this is probably pretty close. (Thanks, @mikhailshilkov!) I'll take a closer look at it tomorrow.

**Comment by praneetloke** [comment]
To be clear, this PR does have the appropriate linking to the generated docs. I was just wondering if we should not link to it until we have #2585 in place as well. But I'll defer that to you.

**Comment by mikhailshilkov** [comment]
> clicking on the "Improve this Doc" links navigates to some weird URL. Should we do something to enable that functionality? Or perhaps disable it?

Nice catch! I looked at what it's supposed to do and it seems that it won't have any value for us. I did not find a way to disable it with configuration while keeping `View Source` button, so I modified the HTML template to remove `Improve this Doc`.

**Comment by praneetloke** [comment]
Ping.

File: docfx/pulumi-template/ManagedReference.common.js [hunk 1]
```diff
@@ -0,0 +1,258 @@
+// Copyright (c) Microsoft. All rights reserved. Licensed under the MIT license. See LICENSE file in the project root for full license information.
```
Comment (lukehoban) [inline docfx/pulumi-template/ManagedReference.common.js]: It's a bit of a shame we have to copy in all these files - are they not available somewhere we can reference instead?  If vendoring them like this recommended?

Assuming we haven't made any changes - perhaps add a README in the folder with details about where this all comes from?

File: docfx/pulumi-template/layout/_master.tmpl [hunk 2]
```diff
@@ -0,0 +1,55 @@
+{{!Copyright (c) Microsoft. All rights reserved. Licensed under the MIT license. See LICENSE file in the project root for full license information.}}
```
Comment (lukehoban) [inline docfx/pulumi-template/layout/_master.tmpl]: If this is for our code - should we add our Copyright instead?

File: docfx/pulumi-template/ManagedReference.common.js (hunk 1 above)
Comment (mikhailshilkov) [reply docfx/pulumi-template/ManagedReference.common.js]: Great point! I found out that we can do overrides with themes, so I removed all files that were untouched and left the ones I changed.

File: docfx/pulumi-template/layout/_master.tmpl (hunk 2 above)
Comment (mikhailshilkov) [reply docfx/pulumi-template/layout/_master.tmpl]: It looks like we don't have copyright in our templates/styles in other parts of docs, so I just removed those top lines. Let me know if it's a wrong thing to do.

**Pull request merged**